src/MongoDB.Driver/FilteredMongoCollectionBase.cs

```diff
@@ -157,6 +182,16 @@ public override Task<IAsyncCursor<TField>> DistinctAsync<TField>(IClientSessionH
             return _wrappedCollection.DistinctAsync(session, field, CombineFilters(filter), options, cancellationToken);
         }
 
+        public override long EstimatedDocumentCount(EstimatedDocumentCountOptions options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            throw new NotSupportedException("EstimatedDocumentCount is not supported for filtered collections.");
```

Comment:
EstimatedDocumentCount can't be supported for filtered collections because a filtered collection has a filter and EstimatedDocumentCount only works with NO filter.